src/web/app/user-join-page.component.ts

```diff
@@ -97,9 +99,15 @@ export class UserJoinPageComponent implements OnInit {
     this.courseService.joinCourse(this.key, this.entityType).subscribe(() => {
       this.navigationService.navigateByURL(this.router, `/web/${this.entityType}`);
     }, (resp: ErrorMessageOutput) => {
-      const modalRef: any = this.ngbModal.open(ErrorReportComponent);
-      modalRef.componentInstance.requestId = resp.error.requestId;
-      modalRef.componentInstance.errorMessage = resp.error.message;
+      const errorMessage = resp.error.message;
+
+      if (errorMessage.match(/Course .* is deleted/)) {
+        this.statusMessageService.showErrorToast(errorMessage);
```

Comment:
A modal might be the better option here since we want to hold the user's attention in this case (critical unrecoverable error). Toasts are really more appropriate for error messages that are not as critical so the user can still use the page without breaking flow.

Side note, don't really like the way we use the error message to check for this error as it's error prone. It's an unexpected way of checking that will be easily missed if someone changes the error message on the backend. I can't really think of a suitable alternative for now though.

Reply:
>@damithc Could we have your input on whether we should use a modal or a toast message?

I prefer a modal as well.

>@samuelfangjw yes I agree that this is not the cleanest solution. The reason is that we need a different behaviour for this specific error, so an alternative is not to throw an error but return a message, either way we still need to use regex to check

Perhaps reconsider using a specific exception for this case?

Reply:
@damithc the frontend can only distinguish exceptions from the backend with status code, so I think using error message is more simple and easier to understand